core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -468,11 +492,10 @@ import scala.util.control.NonFatal
             log.warning("Kafka commit took longer than `commit-time-warning`: {} ms", duration / 1000000L)
           }
           commitsInProgress -= 1
-          if (exception != null) reply ! Status.Failure(exception)
+          if (exception != null) sendReply(Status.Failure(exception))
           else {
-            val committed = Committed(offsets.asScala.toMap)
-            self ! committed
-            reply ! committed
+            self ! Committed(offsets.asScala.toMap)
```

Comment:
// 1. commitAsync callback may not be called at the same thread
    // 2. self is stored in the thread local of object ActorCell
    // so, use self in async callback is wrong. use local variable to save the self ActorRef

Reply:
Thank you for divining into it. Same as above applies here. I should add a comment for that.